src/config/config_generator.h

```diff
@@ -0,0 +1,51 @@
+/*GRB*
+
+Gerbera - https://gerbera.io/
+
+    config_generator.h - this file is part of Gerbera.
+
+    Copyright (C) 2016-2018 Gerbera Contributors
+
+    Gerbera is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License version 2
+    as published by the Free Software Foundation.
+
+    Gerbera is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Gerbera.  If not, see <http://www.gnu.org/licenses/>.
+
+    $Id$
+*/
+
+/// \file config_generator.h
+
+#ifndef GERBERA_CONFIG_GENERATOR_H
+#define GERBERA_CONFIG_GENERATOR_H
+
+#include "IConfigGenerator.h"
+
+class ConfigGenerator : public IConfigGenerator {
+ public:
+  ConfigGenerator();
+  ~ConfigGenerator();
+
+  std::string generate(std::string userHome, std::string configDir, std::string prefixDir, std::string magicFile) override;
+  zmm::Ref<mxml::Element> generateServer(std::string userHome, std::string configDir, std::string prefixDir);
```

Comment:
I looked for a suitable XML DOM generation in C++ but could not find anything easier to implement than the current `mxml::Element`.  Ideally this method will have standard library for XML and expose only the **generate** method.